app/webpacker/components/ContactEditProfilePage/EditProfileFormWithWcaId.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from 'react';
+import { Form, Message } from 'semantic-ui-react';
+import ReCAPTCHA from 'react-google-recaptcha';
+import i18n from '../../lib/i18n';
+import { IdWcaSearch } from '../SearchWidget/WcaSearch';
+import SEARCH_MODELS from '../SearchWidget/SearchModel';
+import EditProfileForm from './EditProfileForm';
+import { contactEditProfileActionUrl } from '../../lib/requests/routes.js.erb';
+import Loading from '../Requests/Loading';
+import Errored from '../Requests/Errored';
+import useSaveAction from '../../lib/hooks/useSaveAction';
+
+export default function EditProfileFormWithWcaId({
+  wcaId: loggedInUserWcaId,
+  loggedInUserData,
+  setContactSuccess,
+  recaptchaPublicKey,
+}) {
+  const [contactEmail, setContactEmail] = useState();
+  const [customWcaId, setCustomWcaId] = useState();
+  const [editProfileReason, setEditProfileReason] = useState();
+  const [editedProfileDetails, setEditedProfileDetails] = useState();
+  const [profileDetailsChanged, setProfileDetailsChanged] = useState(false);
```

Comment:
If a const can be calculated, rather than stored as state, then that's almost always better (via `useMemo` if necessary).
```suggestion
  const profileDetailsChanged = editedProfileDetails && !_.isEqual(editedProfileDetails, profileDetails);
```

That doesn't quite work though, since `profileDetails` is stuck in the child. So I would hoist that (the query/const) up to this component - state should always be at the highest level that it's needed, and it's "needed" here to calculate `profileDetailsChanged`.

If hoisting that up would cause problems, then (in my experience) that means that the components aren't properly thought out, and need to be adjusted.

Reply:
Actually it was not easy, so I finally decided to merge both `EditProfileFormWithWcaId` and `EditProfileForm` into `EditProfileForm`. Now, I guess this problem is solved.

The reason why I splitted them was I was trying to make the react file small. But it was too difficult to manage states, so I thought merging them. I hope this is good, let me know if you have any changes.

As usual, thanks a lot for your review and your work in understanding the code and giving relevant comments.